src/real_estate_simulator.py

```diff
@@ -334,84 +322,87 @@ def refresh_model(self, money_factor):
-        simulation.find_years(LOAN, INTEREST_RATE, ANNUAL_PAYMENT)
-        print(f"\nYear loan = {round(simulation.year, 1)}")
-        simulation.display_loan_interest_statistic()
-    elif MODE == 2:
-        print(f"Surface rate price is {monetary_value(SURFACE_RATE_PRICE)}€/m²")
-        print(f"Contribution is {CONTRIBUTION}€")
-        print(f"Monthly payment is {monetary_value(MONTHLY_PAYMENT)}€/month")
-        print(f"Interest rate is {round(100*INTEREST_RATE, 2)}%")
-        print(f"Years of loan is {LOAN_YEARS}")
-
-        simulation = RealEstatePurchaseSimulator()
-        simulation.find_house_size(SURFACE_RATE_PRICE, NOTARY_COST_RATE, CONTRIBUTION,
-                                   ANNUAL_PAYMENT, INTEREST_RATE, LOAN_YEARS)
-
-        print(f"\nSurface = {simulation.newton_result_model}")
-        simulation.display_loan_interest_statistic()
-    elif MODE == 3:
-        print(f"House price {monetary_value(HOUSE_PRICE)}€")
-        print(f"Monthly payment is {monetary_value(MONTHLY_PAYMENT)}€/month")
-        print(f"Interest rate is {round(100*INTEREST_RATE, 2)}%")
-        print(f"Years of loan is {LOAN_YEARS}")
-
-        simulation = RealEstatePurchaseSimulator()
-        simulation.find_contribution(HOUSE_PRICE, NOTARY_COST_RATE, ANNUAL_PAYMENT, INTEREST_RATE, LOAN_YEARS)
-
-        print(f"\nContribution = {simulation.newton_result_model}€")
-        simulation.display_loan_interest_statistic()
-    elif MODE == 4:
-        print(f"House price is {monetary_value(HOUSE_PRICE)}€")
-        print(f"Notary fees is {monetary_value(NOTARY_FEES)}€")
-        print(f"Contribution is {monetary_value(CONTRIBUTION)}€")
-        print(f"Loan is {monetary_value(LOAN)}€")
-        print(f"Interest rate is {round(100*INTEREST_RATE, 2)}%")
-        print(f"Years of loan is {LOAN_YEARS}")
-
-        simulation = RealEstatePurchaseSimulator()
-        simulation.find_monthly_payment(LOAN, INTEREST_RATE, LOAN_YEARS)
-
-        print(f"\nMonthly payment = {simulation.newton_result_model}€/month")
-        simulation.display_loan_interest_statistic()
-    elif MODE == 5:
-        print(f"House price is {monetary_value(HOUSE_PRICE)}€")
-        print(f"Notary fees is {monetary_value(NOTARY_FEES)}€")
-        print(f"Contribution is {monetary_value(CONTRIBUTION)}€")
-        print(f"Loan is {monetary_value(LOAN)}€")
-        print(f"Monthly payment is {monetary_value(MONTHLY_PAYMENT)}€/month")
-        print(f"Years of loan is {LOAN_YEARS}")
-
-        simulation = RealEstatePurchaseSimulator()
-        simulation.find_interest_rate(LOAN, ANNUAL_PAYMENT, LOAN_YEARS)
-
-        print(f"\nInterest rate = {round(100*simulation.newton_result_model, 3)}%")
-        print(f"\nCumulated interest = {simulation.interest_payment_cumulated[-1]} €")
-        simulation.display_loan_interest_statistic()
+    # Mode can be "Loan duration", "Surface", "Contribution", "Monthly payment", "Interest rate"
+    MODE = "Loan duration"
```

Comment:
suggestion: pourquoi ne pas partir sur un vrai enum ? 
```suggestion
    class Mode(Enum):
        LOAN_DURATION = auto() (ou autre)
        ...
```

Reply:
La variable MODE sert uniquement à lancer le modèle via __name__ == "__main__". L'IHM appel directement la fonction via son nom mais on pourrait imaginer dans le modèle une fonction de lancement basée sur la variable MODE couplé à un dictionnaire {ENUM: nom_fonction}